src/components/NcButton/NcButton.vue

```diff
@@ -346,6 +477,36 @@ export default {
 			default: false,
 		},
 
+		/**
+		 * The text of the button, for icon only buttons you should at least provide the `aria-label` property.
+		 */
+		text: {
+			type: String,
+			default: '',
+		},
+
+		/**
+		 * Additional classes to inject on the text wrapper to allow for custom styling
+		 *
+		 * @example
+		 * ```vue
+		 * <NcButton text="hello" text-classes="my-button-text" />
+		 * <!-- ... -->
+		 * <style>
+		 * :deep(.my-button-text) {
+		 *     text-align: right;
+		 * }
+		 * </style>
+		 * ```
+		 */
+		textClasses: {
+			type: [String, Array],
+			validator(value) {
+				return typeof value === 'string' || (Array.isArray(value) && value.every((v) => typeof v === 'string'))
+			},
+			default: '',
+		},
```

Comment:
Maybe allow all possible Vue classes types, like Object, array of objects and etc?

Reply:
Even `[{a : true}, 'b', ['c', { d: true }]]` is valid in bot Vue 2 and Vue 3. So we can add `textClasses` as an array item to `class` property in render-function even without `additionalTextClasses`

Reply:
Done :)